hive/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -134,10 +136,11 @@ public void commit(TableMetadata base, TableMetadata metadata) {
                 Integer.MAX_VALUE,
                 storageDescriptor(metadata),
                 Collections.emptyList(),
-                new HashMap<>(),
+                Maps.newHashMap(),
                 null,
                 null,
-                ICEBERG_TABLE_TYPE_VALUE);
+            TableType.EXTERNAL_TABLE.toString());
+        tbl.getParameters().put("EXTERNAL", "TRUE"); // using the external table type also requires this
```

Comment:
@aokolnychyi, this is a new fix to make tables external so that Iceberg manages the underlying location instead of Hive. In the previous version of this PR, using a managed table caused the table location to be moved during rename, which would break Iceberg metadata. You probably want to make sure you're using this change.

Reply:
@rdblue yeah, we had this change internally already to be compatible with the Spark external catalog in 2.4. Thanks!

Reply:
This means the metastore won't delete the location directory when the table is dropped. In general, this seems like the right behavior, since Iceberg data can be anywhere. But for users that are only going through this interface, how is data cleaned up? Because tables are created in the default Hive location based on the name, what happens if we do a drop followed by a create (since the old data still exists)?

Reply:
New table data and old table data can be mixed together without corrupting the table state. The problem is that this would leave data lying around. We could add a `purge` option to drop the data when the table is dropped.

Reply:
Who is responsible for cleaning up data in the Iceberg model? Is that out of scope for the project (assumed that a complementary system handles it)?

Reply:
I think it should be up to the caller whether Iceberg cleans it up or leaves it for a complementary system. We have a janitor process that cleans it up, but we don't want to assume that users will. I'll add this in a follow-up issue.

Reply:
Just to clarify: will `purge` flag indicate whether to delete `location` in HMS or will it look into where the data and metadata actually is? (e.g. `write.folder-storage.path` and `write.metadata.path`, which can change over time)

Reply:
Purge would basically do the same thing as a snapshot expiration and would remove the entire tree of metadata files. I don't think it should delete directories because Iceberg can't necessarily assume that it owns the entire directory structure. Iceberg tables actually use the same storage locations.